x-pack/legacy/plugins/ml/public/data_frame/pages/job_management/components/job_list/transform_messages_pane.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useState, useEffect } from 'react';
+
+import { EuiSpacer, EuiBasicTable } from '@elastic/eui';
+// @ts-ignore
+import { formatDate } from '@elastic/eui/lib/services/format';
+import { i18n } from '@kbn/i18n';
+import { ml } from '../../../../../services/ml_api_service';
+
+const TIME_FORMAT = 'YYYY-MM-DD HH:mm:ss';
+
+interface Props {
+  transformId: string;
+}
+
+export const TransformMessagesPane: React.SFC<Props> = ({ transformId }) => {
+  const [messages, setMessages] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  async function getMessages() {
+    try {
+      const messagesResp = await ml.dataFrame.getTransformAuditMessages(transformId);
+      setMessages(messagesResp);
+      setIsLoading(false);
+    } catch (error) {
+      setIsLoading(false);
+      setErrorMessage(
+        i18n.translate('xpack.ml.dfJobsList.jobDetails.messagesPane.errorMessage', {
+          defaultMessage: 'Messages could not be loaded',
+        })
+      );
+    }
+  }
+
+  useEffect(() => {
+    getMessages();
+  }, []);
```

Comment:
I'm afraid this will load the messages only once and not consider the refresh interval. For example, if the row is expanded showing the messages and you'd start a job, the messages would not be updated. We need to think of a way to subscribe to a refresh.

Suggestion: At the moment `use_refresh_interval.ts` takes in `getJobs()` as a callback and is unfortunately tightly coupled to `getJobs()`. We could instead return an observable from `use_refresh_interval.ts` where we can subscribe both `getJobs()` and `getMessages()`. Let me know what you think, happy to discuss approaches.

Reply:
Good point - I've added `latestUpdate` timestamp that the job list keeps track of on refresh and passes to the messages panel as a prop. Now the messages panel refetches every time that value is updated. 84acd6e14cd4ed54af8fa2b3adfbbcb842d4e20b